### PR TITLE
DEVSOL-1903

### PR DESCRIPTION
**Pull request by giteshk**
This should fix the authentication buttons which are probably broken right now.

### CODE REVIEW

**Comment by ossobuffo** [comment]
This will almost certainly break things badly.

It depends on the first-generation .hbr template, in which the original coder didn't know how to generate a comma-delimited list, and thus put a comma after every element. Then they used jQuery to chop the last character (without checking to see if it was a comma). I discovered this when I fixed the problem the right way, then was looking at "text/xm" (because the L had gotten chopped off).

The current handlebars template has "{{#unless @last}}, {{/unless}}" which means that there is no need to chop off the final comma, which the code in lines 473-475 is trying to do. At the very least, those lines of code have to go away.

**Comment by giteshk** [comment]
@ossobuffo  try to enable Oauth/Basic Auth on one of the SmartDocs Method 
See if the buttons are available with the current master template.

**Comment by cnovak** [comment]
It sounds like we should take in this commit then - this isn't code to fix comma separated issue in the template, but a fix for Oauth/Basic Auth.

**Comment by ossobuffo** [comment]
You're right. This looks remarkably similar to the code that existed to work around the content-type stuff, and I got them confused.

But in fact this is a lame fix to an even lamer problem in the handlebars template. The JS is tacking commas on after every authtype, and then depending on jQuery to chop it back off. Can we at least have the code in model.js make sure that the last character is a comma before chopping it off? That way we can fix this dumb stuff in the handlebars template in the future without running into weirdness.

File: static/js/model.js [hunk 1]
```diff
@@ -467,6 +467,13 @@ Apigee.APIModel.Methods = function() {
    * This method invokes the necessary details for the operation page.
    */
   this.init = function() {
+    // Convert the auth type value as user friendly text.
+    var authTypeElement = jQuery("[data-role='auth-type']");
+    authType = jQuery.trim(authTypeElement.text());
+    if (authType.split(",").length > 1) {
```
Comment (ossobuffo) [inline static/js/model.js:473]: if (authType.substr(-1) == ',') {

**Comment by ossobuffo** [comment]
My proposed fix above went live in Dev Portal as part of this PR:
https://github.com/apigee/devportal-profile/pull/261/files#diff-7a926e8c8032f8bc3b9fa60923b24cc6R436
As a related measure, you might want to look at this commit that was made to the handlebars template:
https://github.com/apigee/devportal-profile/pull/266/files

**Pull request closed without merging**